kedro-datasets/tests/pandas/test_sql_dataset.py

```diff
@@ -419,3 +420,37 @@ def test_create_connection_only_once(self, mocker):
         assert mock_engine.call_count == 2
         assert fourth.engines == first.engines
         assert len(first.engines) == 2
+
+    def test_adapt_mssql_date_params_called(self, mocker):
+        """Test that the adapt_mssql_date_params
+        function is called when mssql backend is used.
+        """
+        mock_adapt_mssql_date_params = mocker.patch(
+            "kedro_datasets.pandas.sql_dataset.SQLQueryDataSet.adapt_mssql_date_params"
+        )
+        mock_engine = mocker.patch("kedro_datasets.pandas.sql_dataset.create_engine")
+        ds = SQLQueryDataSet(sql=SQL_QUERY, credentials=dict(con=MSSQL_CONNECTION))
+        mock_engine.assert_called_once_with(MSSQL_CONNECTION)
+        assert mock_adapt_mssql_date_params.call_count == 1
+        assert len(ds.engines) == 1
+
+    def test_adapt_mssql_date_params(self):
+        """Test that the adapt_mssql_date_params
+        function transforms the params as expected, i.e.
+        making datetime date into the format %Y-%m-%dT%H:%M:%S
+        and ignoring the other values.
```

Comment:
Could you add a test that checks that other formats provided are ignored ?

Reply:
These are the values tested: 

`["2023-01-01", "2023-01-01T20:26", "2023", "test", 1.0, 100]`

Some of them aren't dates. Maybe you had something else in mind @AhdraMeraliQB? :thinking: 

Please let me know. :+1:

Reply:
@yassineAlouini I was thinking more along the lines of testing that the appropriate error is thrown when anything other than a list is passed through. This will also complete the code coverage that is preventing the unit tests from passing.

Reply:
Ah, I see now. Thanks for the clarification. :+1: